Buildings/Fluid/Humidifiers/DXDehumidifier.mo

```diff
@@ -0,0 +1,308 @@
+within Buildings.Fluid.Humidifiers;
+model DXDehumidifier "DX dehumidifier"
+  extends Buildings.Fluid.Interfaces.PartialTwoPort;
+
+  parameter Boolean addPowerToMedium = true
+    "Transfer power and heat to the fluid medium";
+
+  parameter Buildings.Fluid.Humidifiers.Data.Generic per
+    "Performance data"
+    annotation (choicesAllMatching=true,
+      Placement(transformation(extent={{60,80},{80,100}})));
+
+  parameter Modelica.Units.SI.VolumeFlowRate VWat_flow_nominal(
+    final min=0)
+    "Rated water removal rate"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Modelica.Units.SI.MassFlowRate mAir_flow_nominal
+    "Nominal mass flow rate of air in the air stream"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Modelica.Units.SI.PressureDifference dp_nominal(
+    displayUnit="Pa")
+    "Pressure difference"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Real eneFac_nominal(
+    final min=0)
+    "Rated energy factor, in liter/kWh"
+    annotation (Dialog(group="Nominal condition"));
+
+  Modelica.Blocks.Interfaces.BooleanInput uEna
+    "Enable signal"
+    annotation (Placement(transformation(extent={{-140,-66},{-100,-26}}),
+      iconTransformation(extent={{-120,30},{-100,50}})));
+
+  Modelica.Blocks.Interfaces.RealOutput T(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Outlet air medium temperature"
+    annotation (Placement(transformation(extent={{100,40},{140,80}}),
+      iconTransformation(extent={{100,30},{120,50}})));
+
+  Modelica.Blocks.Interfaces.RealOutput P(
+    final unit="W",
+    final quantity="Power")
+    "Power consumption rate"
+    annotation (Placement(transformation(extent={{100,-60},{140,-20}}),
+      iconTransformation(extent={{100,-50},{120,-30}})));
+
+  Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow preHeaFlo
+    "Heat transfer into medium from dehumidifying action"
+    annotation (Placement(transformation(extent={{-10,50},{10,70}})));
+
+  Modelica.Thermal.HeatTransfer.Sensors.HeatFlowSensor heaFloSen
+    "Sensor for measuring heat flow rate into medium during dehumidification"
+    annotation (Placement(transformation(extent={{20,70},{40,50}})));
+
+  Modelica.Thermal.HeatTransfer.Sensors.TemperatureSensor senTem
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{60,50},{80,70}})));
+
+  Modelica.Units.SI.MassFraction XOut
```

Comment:
Do we have to calcuate the `XOut` here? I also see that we have the sensor `senMasFra` and `heaFloSen`. Are they required by the model itself? I see that the values are not used anywhere else.

Reply:
@JayHuLBL Thanks for catching this! I did another review of the code, and realized both `XOut` and `senMasFra` are not required. However, we are using `heaFloSen` to validate the heat transfer of the component within the validation model. It will be useful to identify any errors introduced in the calculation, via CI testing. Please refer to #3788. Thanks!